coffee/shepherd.coffee

```diff
@@ -117,6 +117,9 @@ class Step extends Evented
     @tether = new Tether extend(tetherOpts, @options.tetherOptions)
 
   show: =>
+      @options.beforeShowPromise?()?.then(=> @_show()) ? @_show()
```

Comment:
I understand it's ***supposed*** to do that, but it's not binding to this as it should. I'll put in the time to reproduce the error for you, if you're not okay with merging it as is.